Framework/CurveFitting/src/Algorithms/SplineBackground.cpp

```diff
@@ -182,15 +196,16 @@ double SplineBackground::calculateBinWeight(double errValue) {
     // Regardless of which warning we print we should
     // set the bin weight
     outBinWeight = 0;
-    if (errValue <= 0) {
-      g_log.warning("Spline background found an error value of 0 or less on an unmasked"
-                    " bin. This bin will have no weight during the fitting process");
-    } else {
-      // nan / inf
-      g_log.warning("Spline background found an error value of nan or inf on an"
-                    " unmasked bin. This bin will have no weight during the fitting"
-                    " process");
-    }
+    // Temporarily removed this as the warnings slow the algorithm immensely
```

Comment:
Hadn't spotted this. Must have been something Dan did. In other algorithms I've seen a check like this. Perhaps this might help here:
```
    if (g_log.is(Kernel::Logger::Priority::PRIO_DEBUG)) {
      g_log.debug("message");
    }
```

Reply:
I noticed that the call to SplineBackground from the function \_spline_vanadium_ws_per_detector in abstract_inst.py uses EnableLogging=False so maybe the commenting out isn't needed any more at all. I'll add my suggested if statement in anyway to protect any users who are running SplineBackground across multiple spectra from places other than these POLARIS scripts